cleanlab_studio/studio/studio.py

```diff
@@ -126,7 +134,60 @@ def apply_corrections(self, cleanset_id: str, dataset: Any, keep_excluded: bool
         project_id = api.get_project_of_cleanset(self._api_key, cleanset_id)
         label_column = api.get_label_column_of_project(self._api_key, project_id)
         id_col = api.get_id_column(self._api_key, cleanset_id)
-        if _pyspark_exists and isinstance(dataset, pyspark.sql.DataFrame):
+
+        if _snowflake_exists and isinstance(dataset, snowpark.DataFrame):
```

Comment:
Can we refactor the `apply_corrections()` method and pull out the code to handle the separate dataset types into individual helper methods to avoid having this function continue to grow out of control?